core/src/main/java/org/infinispan/commands/functional/ManyDataCommand.java

```diff
@@ -0,0 +1,10 @@
+package org.infinispan.commands.functional;
+
+import java.util.Set;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public interface ManyDataCommand<K> {
```

Comment:
I agree there's duplicate code and to be honest that class could do with some refactoring to avoid that. But I don't know for sure if your point is about duplication between different functional command visit impls or between functional command visit impls and the others.

In any case, I'm happy for such ManyDataCommand interface to exist if it's really going to reduce some code in that class too.